### DISCUSSION TITLE
Build Python without build SCIp

### DISCUSSION THREAD

**Discussion by cvancrae** [google/or-tools#4369] [Build and configuration issues] [answered]
Hi,
I want to build or-tools for Python with SCIP support but from external build of SCIP.
I use this :
cmake -S. -Bbuild -DBUILD_PYTHON=ON -DBUILD_SAMPLES=OFF -DBUILD_EXAMPLES=OFF -DBUILD_TESTING=OFF -DUSE_SCIP=ON -DBUILD_SCIP=OFF

But it build SCIP.

**Comment by sschnug** [comment]
Some remarks:

afaik
- or-tools builds SCIP not with it's default lp-solver SOPLEX but with or-tools' GLOP lp-solver
- SCIPs lp-solver implementation cannot be configured at runtime and the lp-solver decision is applied at SCIP compile-time

This "might" complicate some things for you iff or-tools does not support this in the build-config (don't know, did not check).

(Another topic might be the selection/compilation+patches of the symmetry-breaking backend)

Maybe quickly check the CMake as well as SCIP sources in or-tools with this information at hand.

------

Do you have a good reason to bring your own SCIP?

**Comment by Mizux** [comment, marked as answer]
By default it is not supported, to make a standalone python package we have to ship with SCIP embedded to follow PEP 513..
ref: <https://peps.python.org/pep-0513/#the-manylinux1-policy>

Here `BUILD_DEPS` is forced to ON if any `BUILD_<LANG>` is enable.
https://github.com/google/or-tools/blob/f676293148341b82b1e20f5b582c11fb68b2143e/CMakeLists.txt#L163-L164

https://github.com/google/or-tools/blob/f676293148341b82b1e20f5b582c11fb68b2143e/CMakeLists.txt#L283-L287
Forced to `ON` (last param) if expression `NOT BUILD_DEPS` is false otherwise default to `OFF`.
see: https://cmake.org/cmake/help/latest/module/CMakeDependentOption.html

BUT you can still hack this code (OSS is good !)
so you may hack this lines to `set(BUILD_SCIP OFF)` or `option(BUILD_SCIP OFF)` instead....

we could replace this code by:
```cmake
if(USE_SCIP)
  if(BUILD_DEPS)
    option(BUILD_SCIP "Build the SCIP dependency Library" ON)
  else()
    option(BUILD_SCIP "Build the SCIP dependency Library" OFF)
  endif()
  message(STATUS "Build SCIP: ${BUILD_SCIP}")
endif()
```
so user still have option to change the default (sound) value....
ref: <https://cmake.org/cmake/help/latest/command/option.html>

**Comment by cvancrae** [comment]
With this modification, I can do exactly what I want. 
I think you should modify the CMake file.
